iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -139,15 +145,18 @@ static LogicalResult verifyScatterOp(ScatterOp op) {
         "mismatch in shape of indices and update value at dim#0");
   }
   auto originalType = op.outputs()[0].getType().cast<ShapedType>();
-  if (originalType.getRank() != updateType.getRank()) {
+  // indexDepth + update dims should match to original dims. The first dim of
+  // update is the number of updates.
+  if (originalType.getRank() != indexDepth + updateType.getRank() - 1) {
```

Comment:
This seems off to me as well. From my understanding
if the indices is of rank `M`, and original is of rank `N`, then updates is of rank `N - M + 1`.